bicycleparameters/models.py

```diff
@@ -771,15 +771,19 @@ def simulate(self, times, initial_conditions, input_func=None,
 
         if input_func is None:
             def eval_rhs(t, x):
-                return A@x
+                x = x.reshape((4, 1))
+                return (A@x).squeeze()
         else:
             def eval_rhs(t, x):
-                return A@x + B@input_func(t, x)
+                u = input_func(t, x).reshape((2, 1))
+                x = x.reshape((4, 1))
+                return (A@x + B@u).squeeze()
```

Comment:
this may not be needed, the issue was that the integrator was jumping over the pulse